### ISSUE TITLE
Archivo is damaged

### ISSUE THREAD

**Issue by twenty-one** [fflewddur/archivo#65]
I was able to download, install and use 1.0. But with 1.0.1, I keep getting this message:

_"Archivo.app" is damaged and can't be opened. You should move it to the Trash. Safari downloaded this file today at 8:47 AM from straylightlabs.net_

I've also tried it with Firefox, but I get the same message (with Firefox substituted for Safari).

Any idea how to fix this? Thank you.

**Comment by TivoMan** [comment]
Yes, I have an idea, but no guarantees...

Go to:
System Preferences --> Click on "Security & Privacy"
Authorize [Click on Lock]
Under "Allow apps downloaded from:" section, select "Anywhere"
Close System Preferences.

Try to open. It might quit on you, but try starting it up again. It should work. 
For me...I'm now having a problem having it find my Tivo.
[You might want to change your settings back to the way they were once you have Archivo up and running]

**Comment by fflewddur** [comment]
@twenty-one : Sounds like something got corrupted with the installation. I would try deleting Archivo from /Applications, rebooting your Mac, then download and install it again. I just tried this myself and can confirm that the download works properly.

@TivoMan : I'm a registered Apple developer and have digitally signed the Archivo.app package, so Archivo will run correctly with Apple's default security settings; you don't need to allow apps from anywhere to run Archivo :)

**Comment by TivoMan** [comment]
@fflewddur : I don't think corruption is the problem. Both twenty-one and I downloaded and installed the app multiple times using different browsers. If it was just a simple corruption issue, wouldn't a second (3rd, 4th...) download fix the problem? Perhaps something went awry when you digitally signed the app.???? My suggestion was based on current experience. After downloading the app. I also got the ""Archivo.app" is damaged and can't be opened. You should move it to the Trash..." error multiple times after multiple downloads. The ONLY way I was able to get the app to install was to change the security settings. If Twenty-one does change the security settings and the app does install would that not suggest something more than just a corruption issue?? 

When the app was installing I noticed that a security window appeared for a few seconds before it gave me the error which led me to the idea that there might be something wrong with the digital signature or something.

I figured you are a registered apple developer and that's why I was scratching my head as to why I had to change the security setting, but that was what I had to do in order for me to install the app.

Now if I can only get the app to see my Tivo...:)

**Comment by fflewddur** [comment]
@TivoMan : I'm not sure why changing the security fixed the problem, that's interesting. You can verify that Archivo is properly signed by opening up Terminal and running the following command: `codesign -dv /Applications/Archivo.app/`

The output should look like this:

```Executable=/Applications/Archivo.app/Contents/MacOS/Archivo
Identifier=net.straylightlabs.archivo
Format=app bundle with Mach-O universal (i386 x86_64)
CodeDirectory v=20200 size=442 flags=0x0(none) hashes=7+4 location=embedded
Signature size=8911
Timestamp=Apr 10, 2016, 7:18:34 PM
Info.plist entries=23
TeamIdentifier=3NKBKZ9Y8S
Sealed Resources version=2 rules=12 files=22
Internal requirements count=1 size=188```

I've seen "Archivo is damaged" error on my Mac during development, and always found that the only way to fix it was to delete the app, reboot, then re-install. Simply re-installing over the old version didn't work for me, deleting it and rebooting before re-installing was necessary. This is the first time I've heard of this problem in the production version, though; I assumed I'd just mucked something up during development :)

**Comment by fflewddur** [comment]
PS: I'm looking into why some people have trouble getting Archivo to find their TiVos.

**Comment by TivoMan** [comment]
@fflewddur : There was only one difference, which might be why we are having difficulty installing the app without having to change the security preferences...

Identifier=net.straylightlabs.archivo
Format=bundle with Mach-O universal (i386 x86_64)
CodeDirectory v=20200 size=310 flags=0x0(none) hashes=7+4 location=embedded
Signature size=8911
Timestamp=Apr 10, 2016, 7:18:34 PM
Info.plist entries=23
TeamIdentifier=3NKBKZ9Y8S
Sealed Resources version=2 rules=12 files=22
Internal requirements count=1 size=188

The codedirectory "size" number is different.

**Comment by fflewddur** [comment]
Thanks for checking that! Just to verify: after you delete /Applications/Archivo.app and reboot, the re-installation also shows the same size discrepancy?

**Comment by twenty-one** [comment]
Sorry to take so long to get back to you. I wanted to let you know that the suggestion to change “Allow apps downloaded from” to Anywhere fixed the problem.

Using 1.0.1 I just downloaded two shows from my Tivo and they worked fine.

Thank you for creating this app. If I can help you somehow (I have a wide range of skills, but I was thinking about writing/proofing your use guide or FAQ) please ask.

Regards,
Matthew

**Comment by fflewddur** [comment]
Awesome, glad to hear it's working! Again, to verify for my own debugging: did deleting the app, rebooting, and re-installing not work for you? If that didn't work and changing the security settings did work, then there's likely something wrong with my packaging and code signing script, and I need to know so I can investigate what the problem might be.

**Comment by twenty-one** [comment]
Yes. That seemed to be the solution to the problem. At first I simply tried to download the update and drag it over the existing version. That gave me the error message message (about the app being damaged). Then a few times I tried  deleting 1.0 and downloading 1.0.1, but each time I received the same error.

**Comment by TivoMan** [comment]
@fflewddur : Yes, after deleting the app and then reinstalling it, the app still gave me the same error and the terminal gave me the same data discrepancy. After changing the security setting, the app would launch giving me the "This app was downloaded..." window. After clicking OK, it would quit on me, but it launched just fine the next time I ran it (other than not being able to find my Tivo).

**Comment by fflewddur** [comment]
Hey, I tried a new app builder for the latest Archivo release (https://github.com/fflewddur/archivo/releases/tag/v1.0.2); could either of you let me know if this version also gives you an "Archivo.app is damaged" error?

**Comment by twenty-one** [comment]
Hi. I’m sorry to tell you that I had the same problem (app is damaged…). But the same solution worked: I simply changed the Security settings to allow apps from anywhere. Of course, this is not a solution that would make anyone feel comfortable.

Regards.

**Comment by TivoMan** [comment]
I'll try to install it this week and report back.

-

**Comment by TivoMan** [comment]
@fflewddur:
I downloaded 1.0.3 version and I still get the "app is damaged..." error. I can only get it to work if I change the security settings. This is what I get when checking the sig:

Executable=/Applications/Archivo.app/Contents/MacOS/Archivo
Identifier=net.straylightlabs.archivo
Format=bundle with Mach-O universal (i386 x86_64)
CodeDirectory v=20200 size=330 flags=0x0(none) hashes=8+4 location=embedded
Signature size=8912
Timestamp=May 2, 2016, 9:02:42 PM
Info.plist entries=23
TeamIdentifier=3NKBKZ9Y8S
Sealed Resources version=2 rules=12 files=22
Internal requirements count=1 size=188

**Comment by fflewddur** [comment]
@TivoMan and @twenty-one: is this still an issue with Archivo 1.1 (available at https://straylightlabs.net/archivo/#download)?

**Comment by TivoMan** [comment]
@fflewddur:

YES (&no)...not the exact problem, but still major issue....

I was able to install, but when I tried to launch it, it first quit on me. When I tried to launch it again, it would not quit, but...I would get the following pop up error window:

"We can't seem to access your network...Archivo encountered a problem when it tried to search for TiVos on your network. This is usually caused by another program on your computer that is blocking the net" [BUTTONS: Report Problem; Close; Try Again]. I would click on Try Again, but it would continue to show the error message.

I had to re-install 1.0.5.

**Comment by fflewddur** [comment]
Archivo 1.1.0 includes the ability to select which network it should search for TiVos on, or, if that still isn't working, manually specify your TiVo's IP address. Did you try either of these options? Both are configurable in the Preferences dialog.
